### PR TITLE
Adds proto3 definition

### LINKED ISSUE #39: proto3 encoding
through various issues, seems proto3 is the preferred binary format for the span v2 model. We have some decisions to make, mostly around parity with ids and ips.

For example, we can choose to encode the hex ids as raw bytes (half the size of them)
We can also choose to encode IP addresses literals as raw bytes (sliding scale for ipv6; always smaller for ipv4)

Main thing is that if we do choose a binary encoding for proto3 ids and ips, there's a chance users will encode IDs incorrectly (ex encode IDs as uppercase hex, or '0x' prefixed or worse base64). There's a chance folks will handle ip raw bytes inconsistently (ex we consistently marshal ipv6 format so exact match always works).

thoughts? @openzipkin/core

### PR DESCRIPTION
**Pull request by adriancole**
Design notes:

This is just like json, except uses bytes for IDs and IPs and enum instead of string for span kind

When finished will fix #39

### CODE REVIEW

**Comment by adriancole** [comment]
Well it is just the same ordering as we write json. So more about
consistency than anything else. Can write a note though.. sure!

File: zipkin2.proto [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  // 4 bytes
+  bytes ipv4 = 1;
+  // 16 bytes
+  bytes ipv6 = 3;
+  // technically uint16 would work, but there is none
+  int32 port = 2;
+}
+
+message Annotation {
+  uint64 timestamp = 5;
```
[the remark this answers is not in the document]
Comment (adriancole) [reply zipkin2.proto]: heh this is a mistake!

**Comment by adriancole** [comment]
Added some details about defaults due to gitter chat with @jorgheymans

Also switched to fixed64 for timestamps. Epoch timestamp are encoded fixed64 are varint would also be 8 bytes, and more expensive to encode and size. Duration is stored uint64, as often the numbers are quite small.

**Comment by adriancole** [comment]
on the fixed64 for timestamps thing:

one benefit is constant time for calculation (returning 8 vs looping through 7 bits at a time)
the other is the much faster encoding, for example, the below compares quickest varint vs fixed encode of an epoch micros timestamp into a byte array:

```
Benchmark                              Mode  Cnt  Score    Error  Units
BufferBenchmarks.writeFixed64          avgt   15  0.004 ±  0.001  us/op
BufferBenchmarks.writeVarint64_loop    avgt   15  0.010 ±  0.001  us/op
```

**Comment by adriancole** [comment]
https://github.com/openzipkin/zipkin/pull/1998 shows incremental progress implementing this

**Comment by adriancole** [comment]
@openzipkin/core FYI I am very confident in the shape of this at the moment. There may be details and obviously will lift the docs from the openapi spec, but please do chime in if you see any problems

**Comment by basvanbeek** [comment]
Maybe add comments to expected usage of DependencyLink?

**Comment by jcchavezs** [comment]
LGTM

**Comment by adriancole** [comment]
>
> Maybe add comments to expected usage of DependencyLink?
>
in fact, I'll remove it. it isn't necessary and we have no upload api for
them.

File: zipkin2.proto [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    KIND_UNSPECIFIED = 0;
+    CLIENT = 1;
+    SERVER = 2;
+    PRODUCER = 3;
+    CONSUMER = 4;
+  }
+  Kind kind = 4;
+  string name = 5;
+  fixed64 timestamp = 6;
+  uint64 duration = 7;
```
Comment (yurishkuro) [inline zipkin2.proto]: What are the units for timestamp and duration?

**Comment by adriancole** [comment]
I've migrated docs from the yaml here. All ready to go in my opinion

**Pull request merged**